### PR TITLE
 Remove final_grade from MicromastersCourseCertificate

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Related to #3904, Follow up for #3910 

#### What's this PR do?
Removes the final_grade OneToOne relationship for MicromastersCourseCertificate.

#### How should this be manually tested?
Create two certificate for your user then switch to this branch and run the migrations. Try backwards as well.
To create a certificate for a passed course:
```
from courses.models import *
from grades.models import *
from grades.factories import *
course = Course.objects.filter(title='Digital Learning 200')[0]
course_run = course.courserun_set.all().first()
final_grade = FinalGradeFactory.create(user=user, course_run=course_run, passed=True, status='complete', grade=0.8)
CourseRunGradingStatus.objects.create(course_run=course_run, status='complete')
cert = MicromastersCourseCertificate.objects.create(final_grade=final_grade, user=user, course=course)
```

### CODE REVIEW

**Comment by mbertrand** [comment]
@annagav I get this error when running `migrate`:
```
Running migrations:
  Applying grades.0014_remove_micromasterscoursecertificate_final_grade...Traceback (most recent call last):
  File "manage.py", line 14, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 371, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 365, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 288, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 335, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/commands/migrate.py", line 200, in handle
    fake_initial=fake_initial,
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/migration.py", line 122, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/src/grades/migrations/0014_remove_micromasterscoursecertificate_final_grade.py", line 18, in delete_duplicate_cert
    if not course.has_frozen_runs():
AttributeError: 'Course' object has no attribute 'has_frozen_runs'
```

I think this is because model methods are not available during migrations, a possible workaround is described here: https://stackoverflow.com/questions/28777338/django-migrations-runpython-not-able-to-call-model-methods/37685925

**Comment by annagav** [comment]
@mbertrand I fixed it.

**Comment by annagav** [comment]
@mbertrand I think I fixed the problem

File: grades/migrations/0014_remove_micromasterscoursecertificate_final_grade.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    certificates = MicromastersCourseCertificate.objects.filter(course__program__financial_aid_availability=True)
+    for certificate in certificates:
+        final_grade = FinalGrade.objects.filter(
+            user=certificate.user,
+            course_run__course=certificate.course,
+            passed=True
+        ).order_by('-grade').first()
+        certificate.final_grade = final_grade
+        final_grade.save()
```
Comment (mbertrand) [inline grades/migrations/0014_remove_micromasterscoursecertificate_final_grade.py]: The reverse migration works if I change this line to `certificate.save()`

**Pull request merged**